docs/testplan/Smartswitch-test-plan.md

```diff
@@ -645,6 +672,672 @@ root@sonic:/home/cisco# show chassis modules status
+Name    ID    Oper-Status    State-Detail             State-Value    Time               Reason
+------  ----  -------------  -----------------------  -------------  -----------------  --------------------------------------------------------------------------------------------------------------------------
+DPU0    0     Online         dpu_midplane_link_state  UP             20241003 17:57:22  INTERNAL-MGMT : admin state - UP, oper_state - UP, status - OK, HOST-MGMT : admin state - UP, oper_state - UP, status - OK
+                             dpu_control_plane_state  UP             20241003 17:57:22  All containers are up and running, host-ethlink-status: Uplink1/1 is UP
+                             dpu_data_plane_state     UP             20241001 19:54:30  DPU container named polaris is running, pdsagent running : OK, pciemgrd running : OK
+root@MtFuji:/home/cisco# 
+
+root@sonic:/home/admin# pdsctl show system --events
+----------------------------------------------------------------------------------------------------
+Event                                             Severity  Timestamp                               
+----------------------------------------------------------------------------------------------------
+DSE_SERVICE_STARTED                               DEBUG     2024-09-20 21:48:11.515551 +0000 UTC    
+DSE_SERVICE_STARTED                               DEBUG     2024-09-20 21:48:11.668685 +0000 UTC    
+DSE_SERVICE_STARTED                               DEBUG     2024-09-20 21:48:12.379261 +0000 UTC    
+DSE_SERVICE_STARTED                               DEBUG     2024-09-20 21:48:19.379819 +0000 UTC        
+root@sonic:/home/admin#
+
+```
+
+#### Pass/Fail Criteria
+
+ * Verify that used memory should not cross the specified threshold value (90) of total memory.
+ * Threshold can be set different based on platform.
+ * Verify that dpu_control_plane_state is up under system-health dpu <DPU_NUM> cli.
+ * Verify no memory related events (MEM_FAILURE_EVENT) under pdsctl show system --events cli. This is vendor specific event montioring cli.
+ * Increase the memory to go beyond threshold (head -c <MEM_SIZE> /dev/zero | tail &) and verify  it in pdsctl show system --events cli.   
+
+
+### 1.14 Check DPU status and pcie Link after memory exhaustion on Switch
+
+#### Steps
+
+   * Use `sudo swapoff -a`. Swapping is turned off so the OOM is triggered in a shorter time.
+   * Use 'nohup bash -c "sleep 5 && tail /dev/zero" &' to to run out of memory completely.
+   * It runs on the background and `nohup` is also necessary to protect thebackground process.
+   * Added `sleep 5` to ensure ansible receive the result first.
+   * Check the status and power on DPUs after switch goes for reboot and comes back
+   * Use `show chassis modules status` to check status of the DPUs.
+   * Append to the existing test case: https://github.com/sonic-net/sonic-mgmt/blob/master/tests/platform_tests/test_memory_exhaustion.py
```

Comment:
Can we have a negative test case, where DPU cannot recover gracefully and a hardware reset comes into play?

Reply:
Even this cli does kind of power cycling rather than graceful reboot.